pkg/onchain/datastore.go

```diff
@@ -0,0 +1,91 @@
+package onchain
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	"github.com/massalabs/station/pkg/convert"
+)
+
+type DatastoreContract struct {
+	Data  []byte
+	Args  []byte
+	Coins uint64
+}
+
+type datastoreEntry struct {
+	Key   []byte
+	Value []byte
+}
+
+func argsValue(args []byte) []byte {
+	
+	if len(args) == 0 {
+		return convert.StrToBytes("")
+	}
+
+	return args
+}
+
+/** 
+populateDatastore creates and serializes a datastore for the given contract.
+*/
+func populateDatastore(contract DatastoreContract) ([]byte, error) {
+	var datastore []datastoreEntry
+
+	// nomber of contracts to deploy
+	numberOfContractsKey := []byte{0}
+	numberOfContracts := convert.U64ToBytes(1) 
+	datastore = append(datastore, datastoreEntry{Key: numberOfContractsKey, Value: numberOfContracts})
+
+	// contract data
+	contractKey := convert.U64ToBytes(1) 
+	datastore = append(datastore, datastoreEntry{Key: contractKey, Value: contract.Data})
+
+	// args data 
+	argsKey := []byte{1,0,0,0,0,0,0,0,1,0,0,0,0} // hardcoding as implementation in go of a dynamic key is necessary on if we choose to support multiple SC uploads
+	datastore = append(datastore, datastoreEntry{Key: argsKey, Value: contract.Args})
+
+	// coins data
+	coinsKey := []byte{1,0,0,0,0,0,0,0,1,0,0,0,1} // hardcoding as implementation in go of a dynamic key is necessary on if we choose to support multiple SC uploads
+	datastore = append(datastore, datastoreEntry{Key: coinsKey, Value: convert.U64ToBytes(contract.Coins)})
+
+	// Serialize the datastore
+	serializedDatastore, err := SerializeDatastore(datastore)
+	if err != nil {
+		return nil, err
+	}
+
+	return serializedDatastore, nil
+}
+
+// SerializeDatastore serializes the datastore into a []byte array.
+func SerializeDatastore(datastore []datastoreEntry) ([]byte, error) {
+	var buffer bytes.Buffer
+
+	buf := make([]byte, binary.MaxVarintLen64)
```

Comment:
as you use a go slice, i think you are not forced to declare the size. it should be dynamic

Reply:
it's necessary "Slice: The size specifies the length. The capacity of the slice is
equal to its length. A second integer argument may be provided to
specify a different capacity; it must be no smaller than the
length. For example, make([]int, 0, 10) allocates an underlying array
of size 10 and returns a slice of length 0 and capacity 10 that is
backed by this underlying array." for maps it is not